docs/Tutorials/1_Getting_Started/Getting started.md

```diff
@@ -0,0 +1,207 @@
+
+```Haskell
+aliceRequestCid <- submit alice do
+  createCmd DepositRequest
+    with
+      account = AccountKey with owner = alice; custodian = bank; id = "Alice@Bank"
+      instrument = InstrumentKey with issuer = bank; depository = bank; id = instrumentId
+      amount = 1000.0
+
+aliceCashHoldingCid <- submit bank do exerciseCmd aliceRequestCid CashDepositRequest_Accept
+```
+
+Alice creates a request to deposit 1000 USD at the Bank, the Bank then accepts the request and a corresponding `Holding` is created. ADD LINK TO HOLDING INTERFACE DEFINITION
+
+You can imagine that the latter step happens only after `Alice` has showed up at the bank and delivered physical banknotes corresponding to the deposit's amount.
+
+### Transferring cash from Alice to Bob
+
+The final step of our `Setup` script transfers Alice's holding to Bob using the `Transfer` workflow.
+
+```Haskell
+transferRequestCid <- submit bob do
+  createCmd TransferRequest
+    with
+      receiverAccount = AccountKey with owner = bob; custodian = bank; id = "Bob@Bank"
+      instrument = InstrumentKey with issuer = bank; depository = bank; id = instrumentId
+      amount = 1000.0
+      sender = alice
+
+newHoldingCid <- submit alice do exerciseCmd transferRequestCid CashTransferRequest_Accept with holdingCid = aliceCashHoldingCid
+```
+
+Bob requests the cash to be transferred to his account, Alice accepts the request.
+
+## Further considerations
+
+We now take a look at some aspects of the workflow and try to answer some questions that you might be having.
+
+If you are curious to see some more use-cases for the library, feel free to jump to the next tutorials and come back to this section when you feel it is the right time.
+
+### How does the `Transfer` workflow work?
+
+If you look at the implementation of the `Transfer` workflow, you will notice the following lines
+
+```Haskell
+let transferableCid : ContractId Transferable.I = coerceContractId holdingCid
+
+newTransferableCid <- exercise transferableCid Transferable.Transfer
+  with
+    newOwnerAccount = receiverAccount
+
+pure $ toInterfaceContractId @Holding.I newTransferableCid
+```
+
+The first line casts the holding contract id (of type `ContractId Holding.I`) to the `Transferable` interface using `coerceContractId`.
+
+Then, the `Transfer` choice defined as part of the `Transferable.I` interface is invoked.
+
+Finally, the new holding is cast back to a `Holding.I` before it is returned. This is done using `toInterfaceContractId`.
+
+In order to fully understand these instructions, we need to keep it mind the interface hierarchy used by our holding implementation.
+
+![](interface_hierarchy.png)
+
+We use `coerceContractId` to cast the `Holding` to a `Transferable`. The success of this operation is not guaranteed and will result in a run-time error if the holding implementation at hand does not implement `Transferable`.
```

Comment:
Not sure we should even have `coerceContractId` in our code or even exposing that to users.

Reply:
That's a good point. 

We could hide it behind a helper function, but that means we would needer a helper function for each pair within an interface hierarchy.

I feel that currently it is difficult to work with interface hierarchies without `coerceContractId` (it's used 84 times in the repo).

I will add it as an open point to the corresponding issue.

Reply:
We can use `toInterfaceContractId` [(here)](https://github.com/digital-asset/daml/blob/30f83bae46b3105ee7d7ca0a8de843b3d953b28a/compiler/damlc/daml-stdlib-src/DA/Internal/Interface.daml#L62) and `fromInterfaceContractId` [(here)](https://github.com/digital-asset/daml/blob/30f83bae46b3105ee7d7ca0a8de843b3d953b28a/compiler/damlc/daml-stdlib-src/DA/Internal/Interface.daml#L114) which actually both call `coerceContractId` under the hood.

Reply:
Thanks, I didn't know that both `toInterfaceContractId` and `fromInterfaceContractId` used `coerceContractId` behind the scenes 😅.

At the moment `fromInterfaceContractId` cannot be used because we are missing the `HasFromInterfaceContractId` type-class.

`HasToInterface` and `HasFromInterface` are used to convert templates to interfaces and vice-versa. I believe these are generated automatically be the compiler.

@johan-da leveraged `toInterfaceContractId` to e.g., convert `Account` to `Disclosure`. He did so by
- defining manually `instance HasToInterface Account.I Disclosure.I where _toInterface = asDisclosure`
- `asDisclosure` is defined on the template level as `toInterface @Disclosure.I this` (and uses the `HasToInterface` of the template, which is generated by the compiler)

I don't think we can do the opposite (convert back from `Disclosure` to `Account`) in a similar fashion, as we would need to equip the `Disclosure` interface with an `asAccount` method (that piggybacks on the `HasFromInterface` of the corresponding template). `Disclosure` would need one such a method for each possible interface that might derive disclosure.

Reply:
For all we care we could actually define e.g., `instance HasFromInterface Account Disclosure` with undefined methods. The methods are never called as `fromInterfaceContractId` goes directly to `coerceContractId`.

Reply:
Lets just add your points to a backlog item and we'll have to pick this up with the language team.

Reply:
I added a link to this discussion in the original issue.